docs/sphinx/user_guide/feature/list.rst

```diff
@@ -0,0 +1,256 @@
+
+.. _list-label:
+
+=====
+Lists
+=====
+
+The ``list`` type in the Camp library is a powerful tool for type manipulation and 
+metaprogramming in C++. It allows you to create and operate on sequences of types 
+at compile time, enabling various operations such as accessing, filtering, and 
+transforming types.
+
+Some important notes: 
+
+.. note:: All Camp structures and classes are within the namespace ``camp``.
+
+------------
+Introduction
+------------
+
+In C++, templates provide a way to define functions and classes that operate on types. 
+The ``list`` type serves as a container for types, similar to how a standard container 
+holds values. This allows for advanced type-level programming, where you can perform 
+operations on types as if they were values.
+
+The ``list`` type is defined as a variadic template, allowing you to store any number 
+of types. You can use it to create type sequences, manipulate them, and apply various 
+algorithms to them.
+
+Common use cases for ``list`` include:
+
+* Storing a collection of types for type traits or type transformations.
+* Implementing compile-time algorithms that operate on types.
+* Creating type-safe interfaces or APIs that leverage type information.
+
+----------------------
+Usage and Example Code
+----------------------
+
+The ``list`` structure is literally just a variadic template struct: 
+
+.. code-block:: cpp
+
+    template <typename... Ts>
+    struct list {
+      using type = list;
+    };
+
+Thus, the ``list`` type is incredibly versatile as it can handle any number of types. Since ``list`` is just a structure with a single 
+member ``type``, there are no methods that are directly part of ``list``; however, Camp provides many helper methods that act on lists. 
+Here are some examples demonstrating how to use the ``list`` type and its associated functions. 
+
+The base ``list`` structure, the list-specific ``at`` construct, and the ``find_if`` construct are 
+all included in ``camp/include/camp/list.hpp`` as well as all of the other helper methods described below.  
+
+Creating a List
+^^^^^^^^^^^^^^^
+
+You can create a list of types using the ``list`` template:
+
+.. code-block:: cpp
+
+    using MyList = camp::list<int, double, char>;
+
+Alternatively, you can turn a different type into a list using the ``as_list`` construct: 
+
+.. code-block:: cpp 
+
+    using MyList = camp::as_list</*listable construct*/>
```

Comment:
It would be helpful to have an example here that shows what a "listable construct" is.